File: LV2/Z3.py
from machine import Pin
from time import sleep

LED0 = Pin(12, Pin.OUT)
LED1 = Pin(13, Pin.OUT)
LED2 = Pin(14, Pin.OUT)
LED3 = Pin(15, Pin.OUT)
LED4 = Pin(4, Pin.OUT)
LED5 = Pin(5, Pin.OUT)
LED6 = Pin(6, Pin.OUT)
LED7 = Pin(7, Pin.OUT)

outputLEDs = [LED0, LED1, LED2, LED3, LED4, LED5, LED6, LED7]


def setZeroes():
    for pin in outputLEDs:
        pin.value(0)


def setOnes():
    for pin in outputLEDs:
        pin.value(1)


def runningLight(pauseTime):

    index = -1

    while True:

        while index + 1 < len(outputLEDs):
            outputLEDs[index].off()
            outputLEDs[index + 1].on()
            index += 1
            sleep(pauseTime)

        while index > 0:
            outputLEDs[index - 1].on()
            outputLEDs[index].off()
            index -= 1
            sleep(pauseTime)


def main():
    setOnes()
    sleep(1)
    setZeroes()

    runningLight(0.2)

    print("Uredan kraj!")


if __name__ == "__main__":
    main()

File: LV2/Z3.cpp
#include "mbed.h"
#include "lpc1114etf.h"

DigitalOut E(LED_ACT);

DigitalOut led1(LED0), led2(LED1), led3(LED2), led4(LED3), led5(LED4), led6(LED5), led7(LED6), led8(LED7);

DigitalOut LEDs[] = {led1, led2, led3, led4, led5, led6, led7, led8};

int NUM_LEDS = 8;

void runningLight(int pauseTime_ms) {
    int index = -1;
    while (index + 1 < NUM_LEDS) {
        LEDs[index] = 0;
        LEDs[index + 1] = 1;
        index++;
        wait_us(pauseTime_ms);
    }

    index = NUM_LEDS - 1;

    // Svi LED-ovi su uključeni
    for (int i = 0; i < NUM_LEDS; i++) {
        LEDs[i] = 1;
    }

    wait_us(pauseTime_ms);

    while (index >= 0) {
        LEDs[index] = 0;
        index--;
        wait_us(pauseTime_ms);
    }
}

int main() {
    E = 0;
    while (true)
        runningLight(200000);
}